### PR TITLE
Add MLFlow as a logger option

### PR DESCRIPTION
**Pull request by aspfohl**
Installs mflow by default and allows for mlflow logging!

You just need to specify the mlflow logger in the yaml, this will even work for hosted mlflow:
```
  loggers:
    mlflow:
      experiment_name: /Users/<email>/<experimentfolder>
```

### CODE REVIEW

**Review by eracah — COMMENTED** [review]
Looks pretty good. Can you add `mlflow.log_params(params=hyperparameters)` to the [log_config](https://github.com/mosaicml/llm-foundry/blob/ca0bf0fc83b27d734d53483d2e87fdb60c462a68/llmfoundry/utils/config_utils.py#L91) function similar to how WandB does it

File: llmfoundry/utils/config_utils.py [hunk 1]
```diff
@@ -97,3 +97,11 @@ def log_config(cfg: DictConfig):
             raise e
         if wandb.run:
             wandb.config.update(om.to_container(cfg, resolve=True))
+
+    if 'mlflow' in cfg.get('loggers', {}):
+        try:
+            import mlflow
+        except ImportError as e:
+            raise e
+        if mlflow.active_run():
```
Comment (aspfohl) [inline llmfoundry/utils/config_utils.py:106]: I think [this](https://mlflow.org/docs/latest/tracking.html#logging-functions) is what I want?

Comment (eracah) [reply llmfoundry/utils/config_utils.py:107]: actually, we tested it last week. we're good!

**Review by eracah — APPROVED** [review]
LGTM!

**Pull request merged**